bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BXMLType.java

```diff
@@ -64,4 +73,26 @@ public int getTag() {
     public boolean isAnydata() {
         return true;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!super.equals(obj) || !(obj instanceof BMapType)) {
```

Comment:
What's the use of this `instanceof` check? The `xml` type is not a mapping type.

Reply:
Also, why check for the negation of `equals()`?

Reply:
> What's the use of this `instanceof` check? The `xml` type is not a mapping type.
This is a typo. Fixed that. 
I have followed the implementation BMapType instance coz it already has support for constraint. (Hence the typo) 

> Also, why check for the negation of `equals()`?
The check returns false